test_catch/ParenthesesTests.cpp

```diff
@@ -54,29 +66,543 @@ SCENARIO("Parentheses", "[Parentheses]") {
+            }
+        }
+    }
+    GIVEN("Parentheses, {{[]}{(())}}} input string") {
+        std::string input{"{{[]}{(())}}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == true);
+            }
+        }
+    }
+    GIVEN("Parentheses, {{{()}()}}(){} input string") {
+        std::string input{"{{{()}()}}(){}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == true);
+            }
+        }
+    }
+    GIVEN("Parentheses, {{{}} input string") {
+        std::string input{"{{{}}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == false);
+            }
+        }
+    }
+     GIVEN("Parentheses, {(){()})}}) input string") {
+        std::string input{"{(){()})}})"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == false);
+            }
+        }
+    }
+    GIVEN("Parentheses, {())}}{} input string") {
+        std::string input{"{())}}{}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == false);
+            }
+        }
+    }
+    GIVEN("Parentheses, {()())}} input string") {
+        std::string input{"{()())}}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == false);
+            }
+        }
+    }
+       GIVEN("Parentheses, {(){})}} input string") {
+        std::string input{"{(){})}}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == false);
+            }
+        }
+    }
+      GIVEN("Parentheses, {())}} input string") {
+        std::string input{"{())}}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == false);
+            }
+        }
+    }
+     GIVEN("Parentheses, {{(){{[]{(())}}}}} input string") {
+        std::string input{"{{(){{[]{(())}}}}}"};
+        Parentheses p;
+
+        WHEN("isBalanced() is called") {
+            auto result = p.isBalanced(input);
+
+            THEN("The result is true") {
+                CHECK(result == true);
+            }
+        }
+    }
```

Comment:
Dużo testów sie namnożyło. To dobrze, o ile każdy przypadek jest inny i testuje inny kawałek logiki implementacji. U was chyba jest pewna nadmiarowość. Zredukujcie testy, które niczego nowego nie wnoszą, a podobne testy pogrupujcie używając GENERATE.